src/Microsoft.DotNet.Darc/src/DarcLib/RemoteRepoBase.cs

```diff
@@ -0,0 +1,158 @@
+                        }
+
+                        LibGit2Sharp.Commands.Stage(localRepo, "*");
+
+                        LibGit2Sharp.Signature author = new LibGit2Sharp.Signature(dotnetMaestro, $"@{dotnetMaestro}", DateTime.Now);
+                        LibGit2Sharp.Signature commiter = author;
+                        localRepo.Commit(commitMessage, author, commiter, new LibGit2Sharp.CommitOptions
+                        {
+                            AllowEmptyCommit = false,
+                            PrettifyMessage = true
+                        });
+
+                        localRepo.Network.Push(localRepo.Branches[branch], new LibGit2Sharp.PushOptions
+                        {
+                            CredentialsProvider = (url, user, cred) =>
+                            new LibGit2Sharp.UsernamePasswordCredentials
+                            {
+                                // The PAT is actually the only thing that matters here, the username
+                                // will be ignored.
+                                Username = dotnetMaestro,
+                                Password = pat
+                            }
+                        });
+                    }
+                }
+                catch (LibGit2Sharp.EmptyCommitException)
+                {
+                    _logger.LogInformation("There was nothing to commit...");
+                }
+                catch (Exception exc)
+                {
+                    throw new DarcException($"Something went wrong when pushing the files to repo {repoUri} in branch {branch}", exc);
+                }
+                finally
+                {
+
+                    try
+                    {
+                        // Libgit2Sharp behaves similarly to git and marks files under the .git/objects hierarchy as read-only, 
+                        // thus if the read-only attribute is not unset an UnauthorizedAccessException is thrown.
+                        GitFileManager.NormalizeAttributes(tempRepoFolder);
+
+                        Directory.Delete(tempRepoFolder, true);
+                    }
+                    catch (DirectoryNotFoundException)
+                    {
+                        // If the directory wasn't found, that means that the clone operation above failed
+                        // but this error isn't interesting at all.
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Decode a base64 string or return the original string if not in base 64 form.
+        /// </summary>
+        /// <param name="encodedContent">Potentially base64 encoded string</param>
+        /// <returns>Decoded string as utf8</returns>
+        protected string GetDecodedContent(string encodedContent)
+        {
+            try
+            {
+                byte[] content = Convert.FromBase64String(encodedContent);
+                return Encoding.UTF8.GetString(content);
```

Comment:
>Encoding.UTF8.GetString(content); [](start = 23, length = 33)

And then why do we assume whatever inside the base64 is then UTF-8 text? This method is rather confusing...

Reply:
IIRC, the text returned from the github/azdo APIs is always utf8 (